test/functional/viml/completion_spec.lua

```diff
@@ -100,4 +100,43 @@ describe('completion', function()
       eq('', eval('getline(3)'))
     end)
   end)
+  describe('with always option', function ()
+    before_each(function ()
+      source([[
+function! TestCompletion(findstart, base) abort
+  if a:findstart
+    let line = getline('.')
+    let start = col('.') - 1
+    while start > 0 && line[start - 1] =~ '\a'
+      let start -= 1
+    endwhile
+    return start
+  else
+    let ret = []
+    for m in split("January February March April May June July Auguest September October November December")
+      if m =~ a:base            " match by regex
+        call add(ret, m)
+      endif
+    endfor
+    return {'words':ret, 'refresh':'always'}
+  endif
+endfunction
+
+set completeopt=menuone,noselect
+set completefunc=TestCompletion
+      ]])
+    end )
+
+    it('should complete when add more char', function ()
+      -- to select first word after input char:
+      -- <Down><C-y> work, <C-n> not work.
+      -- but <C-n><C-n>work. there may have some bugs with <C-n>
+      feed('i<C-x><C-u>gu<Down><C-y><ESC>')
```

Comment:
A single `<C-n>` _should_ work here (and it does, when running interactively). But for some reason `<C-n><C-n>` (or `<down><c-y>` as you noted) is needed in the test.

Going to chalk this up to a test helper issue. I didn't find any regressions in normal usage.